### PR TITLE
Detect if a Dapp supports EIP-6963

### PR DESCRIPTION
**Pull request by everdimension**

### CODE REVIEW

File: src/content-script/eip6963.ts [hunk 1]
```diff
@@ -15,19 +15,33 @@ const info = {
...
 
-type Options = { onRequestProvider: () => void };
+type Options = {
+  onRequestProvider?: () => void;
+  onAccessProvider?: () => void;
+};
 
 export function initializeEIP6963(
   provider: EthereumProvider,
   options?: Options
```
Comment (zerts) [inline src/content-script/eip6963.ts:25]: Do we need these options to be optional?

File: src/content-script/in-page.ts [hunk 2]
```diff
@@ -145,8 +145,12 @@ if (dappsWithoutCorrectEIP1193Support.has(window.location.origin)) {
 
 initializeEIP6963(provider, {
   onRequestProvider: () => {
-    // DApp supports EIP-6963
     pageObserver.stop();
+    dappDetection.registerEip6963SupportOnce(provider);
+  },
+  onAccessProvider: () => {
```
Comment (zerts) [inline src/content-script/in-page.ts:151]: Do we need to keep 2 different callbacks?

File: src/content-script/eip6963.ts (hunk 1 above)
Comment (everdimension) [reply src/content-script/eip6963.ts:25]: I think it's a good practice, why not?

File: src/content-script/in-page.ts (hunk 2 above)
Comment (everdimension) [reply src/content-script/in-page.ts:151]: They have different semantic, I think it's better to be truthful with intentions. Calling "requestProvider" one more time when it's not actually requested can be both unexpected and undesired

**Pull request merged**